### PR TITLE
Restore returning selected channels in gain selection

### PR DESCRIPTION
**Pull request by watsonjj**

### CODE REVIEW

**Comment by vuillaut** [comment]
That was quick.
Thank you.

File: ctapipe/calib/camera/gainselection.py [hunk 1]
```diff
@@ -45,14 +45,17 @@ def __call__(self, waveforms):
             Shape: (n_pix, n_samples)
         """
         if waveforms.ndim == 2:  # Return if already gain selected
-            return waveforms
+            pixel_channel = None  # Provided by EventSource
```
Comment (kosack) [inline ctapipe/calib/camera/gainselection.py]: is `pixel_channel` defined?

Comment (kosack) [reply ctapipe/calib/camera/gainselection.py]: never mind - seems it was fixed as I typed that :)

**Comment by vuillaut** [comment]
Ha didn't pay attention enough.
`pixel_channel` shouldn't be of type `bool` rather than `int`?

**Comment by watsonjj** [comment]
> Ha didn't pay attention enough.
> `pixel_channel` shouldn't be of type `bool` rather than `int`?

I don't think so. Then it would be used as a mask instead of an index when indexing numpy arrays

**Comment by vuillaut** [comment]
Ha yes thanks, it's not a mask indeed.

**Pull request merged**